analyzer/src/main/kotlin/managers/PIP.kt

```diff
@@ -153,6 +154,18 @@ class PIP(name: String, analyzerConfig: AnalyzerConfiguration, repoConfig: Repos
             "pypi.org",
             "pypi.python.org" // Legacy
         ).flatMap { listOf("--trusted-host", it) }.toTypedArray()
+
+        /**
+         * Return a version string with leading zeros of components stripped.
+         */
+        fun stripLeadingZerosFromVersion(version: String) =
```

Comment:
Does this `version` parameter only contain digits and dots or are there valid cases where it contains additional characters. E.g. for artifacts "${name}-${version}-{classifier}" with non empty classifiers?

An option could be to: try/catch for each subsequence (between two dots) separately.

Reply:
The version string surely may contain other characters than digits and dots, see https://www.python.org/dev/peps/pep-0440/. But that's nothing the current implementation should struggle with...

> An option could be to: try/catch for each subsequence (between two dots) separately.

That's what the code already does. I guess you're confused by `split(".").joinToString(".")` which at first looks like a no-op, but the code following that line is an implicit mapping on the individual parts between two dots.

Reply:
I see, silly me - sorry then